### PR TITLE
[ENGG-1928] feat: Modheader importer

### PR DESCRIPTION
**Pull request by nafees87n**
Closes issue: 

## ✅ Checklist:

- [ ] Make sure linting and unit tests pass.
- [ ] No install/build warnings introduced.
- [ ] Verified UI in browser.
- [ ] For UI changes, added/updated analytics events (if applicable).
- [ ] For changes in extension's code, manually tested in Chrome and Firefox.
- [ ] For changes in extension's code, both MV2 and MV3 are covered.
- [ ] Added/updated unit tests for this change.
- [ ] Raised pull request to update corresponding documentation (if already exists).

### CODE REVIEW

File: app/src/features/rules/screens/rulesList/components/RulesList/components/ModheaderImporter/ModheaderImporter.tsx [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+              onFilesDrop(files);
+            }}
+            isProcessing={isDataProcessing}
+            title="Drag and drop your Modheader export file to upload"
+            subtitle="Accepted file formats: json"
+            selectorButtonTitle={isParseComplete || validationError ? "Try another file" : "Select file"}
+          />
+        )}
+
+        {(isParseComplete || validationError) && (
```
Comment (RuntimeTerror10) [inline app/src/features/rules/screens/rulesList/components/RulesList/components/ModheaderImporter/ModheaderImporter.tsx:182]: There is a same condition block at line 183. Both the markups can be accommodated in single `(isParseComplete || validationError)` block.

**Comment by RuntimeTerror10** [comment]
The Modheader import modal shows successfully parsed message when importing a Requestly's rules export JSON file


https://github.com/user-attachments/assets/43a9aec4-bf2b-4450-9b66-6bdf4366bd82

**Comment by RuntimeTerror10** [comment]
`CharlesImporter` and `ModheaderImporter` components have almost identical stylesheets as well as the import view where only copy and import action is different.

Can we create a common stylesheet for the importer screens so that any style changes made will be reflected on both the components. And for the import view , can we create a component `ImportView`, here we can pass the copy and import action of the respective app.

**Comment by wrongsahil** [comment]
Fixed #1642

**Pull request merged**